staging_vespalib/src/vespa/vespalib/util/shutdownguard.cpp

```diff
@@ -10,36 +10,30 @@ namespace vespalib {
 
 namespace {
 enum { STACK_SIZE = (1u << 16) };
-
-uint64_t getTimeInMillis() {
-    struct timeval mytime;
-    gettimeofday(&mytime, 0);
-    uint64_t mult = 1000;
-    return (mytime.tv_sec * mult) + (mytime.tv_usec / mult);
-}
 }
 void ShutdownGuard::Run(FastOS_ThreadInterface *, void *)
 {
-    while (_dieAtTime > getTimeInMillis()) {
-        FastOS_Thread::Sleep(5);
+    while (_dieAtTime > steady_clock::now() && ! GetThread()->GetBreakFlag()) {
+        std::this_thread::sleep_for(5ms);
     }
-    if (_dieAtTime != 0) {
+    if (_dieAtTime < steady_clock::now()) {
```

Comment:
Consider making this `<=` in the off-chance that `_dieAtTime == X` and `steady_clock::now()` returns `X` both in the while-loop and in the if-statement. Would otherwise be `!(X > X)` for the former (exiting the loop) and `!(X < X)` in the latter, which won't trigger the shutdown.

Reply:
Good point, fixed